enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -180,4 +188,71 @@ impl<'a> Handler<'a> {
             ret
         }
     }
+
+    /// Do a readv() syscall
+    pub fn readv(&mut self) -> u64 {
+        let fd = self.aex.gpr.rdi;
+        let iovec = self.aex.gpr.rsi as *mut u8;
+        let iovcnt = self.aex.gpr.rdx as usize;
+
+        // Add up total size of buffers.
+        let iovec = unsafe { from_raw_parts_mut(iovec as *mut Iovec, iovcnt) }; // ???
+        let mut bufsize: usize = 0;
+        for i in 0..iovcnt {
+            bufsize += iovec[i].size;
+        }
+
+        // Allocate some unencrypted memory.
+        let size = bufsize + size_of::<Iovec>() * iovcnt;
+        let map = match self.ualloc(size as u64) {
+            Err(errno) => return errno.into_syscall(),
+            Ok(map) => map,
+        };
+
+        // Cast beginning of allocated memory as iovec slice of length iovcnt.
+        let map = unsafe { from_raw_parts_mut(map as *mut Iovec, iovcnt) };
```

Comment:
```suggestion
        let untrusted = unsafe { from_raw_parts_mut(map as *mut Iovec, iovcnt) };
```

Reply:
The general strategy with my naming here is to always communicate to future readers whether the types are trusted or not. We want to avoid accidents...